[stm32f103rg] Device file broken

When enabling stm32f103rg in the SConscript.check many warnings of this kind occur:

```
/Users/usr/Dev/2017amt/xpcc_feature_stm32f107/build/release/test/stm32/libxpcc/src/xpcc/architecture/../../../generated_platform/driver/gpio/stm32f1/gpio.hpp:1531:50: warning: left shift count >= width of type [-Wshift-count-overflow]
   AFIO->MAPR = (AFIO->MAPR & ~(1 << 37)) | (0 << 37);
                                                  ^~
```
To me, it suggests that the [device file](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/devices/stm32/stm32f103-r_v_z-f_g.xml#L116) is broken.

@salkinium Can you please confirm?

--- a/SConscript.check
+++ b/SConscript.check
@@ -97,7 +97,7 @@ devices = [
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f103t4',	'clock':  72000000 },	# low density
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f103rb',	'clock':  72000000 },	# medium density
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f103ze',	'clock':  72000000 },	# high density
-#	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f103rg',	'clock':  72000000 },	# xl density
+	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f103rg',	'clock':  72000000 },	# xl density
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f107vc',	'clock':  72000000 },	# connectivity
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f205ve',	'clock': 120000000 },
 #	{ 'path': 'release/tests/stm32',	'arch': 'cortex-m3',	'device': 'stm32f207ze',	'clock': 120000000 },

--- a/src/xpcc/architecture/platform/driver/gpio/stm32f1/gpio.hpp.in
+++ b/src/xpcc/architecture/platform/driver/gpio/stm32f1/gpio.hpp.in
@@ -306,7 +306,8 @@ public:
 				%% set afpos = af_all[0]
 				%% set afmask = af_all[1]
 				%% set afval = af_all[2]
-				%% set enableAlternateFunction = "AFIO->MAPR = (AFIO->MAPR & ~(" ~ afmask ~ " << " ~ afpos ~ ")) | (" ~ afval ~ " << " ~ afpos ~ ");"
+				%% set reg = "MAPR" if (afpos | int) < 32 else "MAPR2"
+				%% set enableAlternateFunction = "AFIO->" ~ reg ~ " = (AFIO->" ~ reg ~ " & ~(" ~ afmask ~ " << " ~ (afpos | int) % 32 ~ ")) | (" ~ afval ~ " << " ~ (afpos | int) % 32 ~ ");"
 			%% endif
 
 			%% set use = not (	(af.type == "in"     and type == "Output") or

